src/main/java/no/unit/nva/doi/fetch/service/DoiProxyService.java

```diff
@@ -76,10 +78,10 @@ public DoiProxyResponse lookup(URL doiUrl, String apiUrl, String authorization)
     }
 
     private Request validateRequest(Request request) throws MalformedRequestException {
-        if (request.doi == null) {
-            throw new MalformedRequestException(REQUEST_BODY_MALFORMED);
+        if (DoiValidator.validate(request.getDoi())) {
+            return request;
         }
-        return request;
+        throw new MalformedRequestException(REQUEST_BODY_MALFORMED);
```

Comment:
Are there are errors that throw a MalformedRequestException? Maybe better a specialized MalformedDoiExpcetion?